src/model/error.mli

```diff
@@ -1,5 +1,9 @@
 type t
 
+type 'a with_warnings = { value : 'a; warnings : t list }
+
+type warning_accumulator
+
```

Comment:
Why these changes ?

Reply:
Oops, I think I meandered off into 'separate out the parser library' territory while doing this work, and didn't quite revert everything!